android/src/main/java/com/adyenreactnativesdk/component/base/BaseModule.kt

```diff
@@ -86,10 +87,45 @@ abstract class BaseModule(context: ReactApplicationContext?) : ReactContextBaseJ
                 ?: throw Exception("Not an AppCompact Activity")
         }
 
+    open fun getRedirectUrl(): String {
+        return getReturnUrl(reactApplicationContext)
+    }
+
+    override fun onSubmit(state: PaymentComponentState<*>) {
+        var extra: JSONObject? = null
+        if (state is GooglePayComponentState) {
```

Comment:
I am wondering if it is possible to have the google pay component handling inside the google pay module instead of the base module. Can the google pay module override onSubmit as well?